prime-router/src/main/kotlin/Report.kt

```diff
@@ -833,6 +835,38 @@ class Report : Logging {
         )
     }
 
+    /**
+     * Gets the item hash for a the [rowNum] of the report. Returns the ByteArray hash.
+     */
+    fun getItemHashForRow(rowNum: Int): String {
+        // calculate and store item hash for deduplication purposes for the generated item
+        val row = this.table.row(rowNum)
+        var rawStr = ""
+        for (colNum in 0 until row.columnCount()) {
+            rawStr += row.getString(colNum)
+        }
+
+        // combine collectionTime and digest for lower chance of collision
+        val digest = MessageDigest
+            .getInstance("SHA-256")
+            .digest(rawStr.toByteArray())
+
+        return DatatypeConverter.printHexBinary(digest).uppercase()
+    }
+
+    /**
+     * Clears existing table and re-adds the rows that are not being removed
+     */
+    fun removeRowsFromTable(removeRows: List<Int>) {
+        var existingTable = this.table.copy()
+        this.table.clear()
+        for (rowNum in 0 until existingTable.rowCount()) {
+            if (!removeRows.contains(rowNum)) {
+                this.table.append(existingTable.row(rowNum))
```

Comment:
Instead of looping, you could just do a filterIndexed. Something like this might work (provided the table implements the right iterable interfaces:

```
existingTable.filterIndexed{ index, _ -> index !in removeRows }.forEach { row -> this.table.append(row) }
```